src/test/deploy/functions/validate.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { expect } from "chai";
+import * as fsutils from "../../../fsutils";
+import * as validate from "../../../deploy/functions/validate";
+import * as projectPath from "../../../projectPath";
+import * as FirebaseError from "../../../error";
+import * as sinon from "sinon";
```

Comment:
rather than using `sinon.restore()` in all the tests, I'd recommend making a sandbox. It works the same as the module directly but allows for some better grouping behavior.

```typescript
const sandbox = sinon.sandbox();
// ...
sandbox.stub(module, "function");
// ...
sandbox.restore(); // Restores all functions!
```

Reply:
added sandbox for all tests that use sinon